packages/react-components/react-motions-preview/src/utils/groups/getNextChildMapping.ts

```diff
@@ -0,0 +1,44 @@
+import { mergeChildMappings } from './mergeChildMappings';
+import type { PresenceGroupChildMapping } from './types';
+
+export function getNextChildMapping(
+  prevChildMapping: PresenceGroupChildMapping,
+  nextChildMapping: PresenceGroupChildMapping,
+) {
+  const childrenMapping = mergeChildMappings(prevChildMapping, nextChildMapping);
+
+  Object.entries(childrenMapping).forEach(([key, childDefinition]) => {
+    const hasPrev = key in prevChildMapping;
+    const hasNext = key in nextChildMapping;
+
+    const prevChild = prevChildMapping[key];
+    const isLeaving = !prevChild?.visible ?? false;
+
+    // item is new (entering)
+    if (hasNext && (!hasPrev || isLeaving)) {
```

Comment:
I don't understand the `hasNext && isLeaving` case, could you please clarify that one for me?

Reply:
I also, I added a few tests and restructured `if` conditions there.